aws/README.md

```diff
@@ -0,0 +1,98 @@
+
+## Development
+
+### One time testing
+
+Commands to use:
+
+- `npm run generate-event <API METHOD> <Function name>`: This generates an event for a function we declared in `template.yaml` e.g. `SearchFunction`, then we can modify the event with different parameters for one-time testing.
+- `npm run invoke <Function name>`: To be used with the `npm run generate-event` above. This will make a call to the API with parameters defined in the corresponding `event.[].local.json`.
+
+Please check out more details on doing one time testing for the existing functions here:
+
+- [Search function](https://github.com/aiven/devportal/blob/feature/use-aws/aws/SEARCH.md)
+- [Creating feedback function](https://github.com/aiven/devportal/blob/feature/use-aws/aws/CreateFeedback.md)
+
+### Continuous testing
+
+Commands to use:
+
+- `npm run watch`: Start a nodemon instance to watch for changes in `js,json,yaml` and run `sam build` when needed.
+- `npm run start`: Run this in another terminal window. This creates a server with configurations declared in `env.local.json` above, so we can test the API with Postman / browser / etc.
+
+During development, we may want to supply our own services e.g. OpenSearch / PostgreSQL / etc. through `env.local.json` so it doesn't conflict with production instances.
+
+### Testing with local documentation site
+
+With the server running, one can also test the API inside the local documentation site. This is done by changing the API URL inside their corresponding `.html` files, like `_templates/search.html` and `feedback-form.html` to match the local API URL.
+(We're currently checking how to make this less manual)
+
+From there, one can run `make livehtml` in the root folder to serve the doc site, and test the functions directly there.
+
+## Production
+
+We use `sam` to help deploy the changes as well. Be noted that if we change the AWS Lambda stack, we also have to change the API URL where it's used e.g. in `feedback-form.html` and in `search.html`.
+
+First, we need to configure our AWS credentials locally so we can deploy (this may change when we use Aiven AWS account) - Open `~/.aws/credentials`, fill:
+
+```
+      aws_access_key_id = <access_key_id>
+      aws_secret_access_key = <secret_access_key>
+```
+
+Both of those can be obtained from [AWS Security Credentials section](https://us-east-1.console.aws.amazon.com/iam/home#/security_credentials$access_key). Be noted that these access keys give unrestricted access to the entire AWS account.
+
+Commands to use:
+
+- `npm run deploy`: This runs a guided version of `sam deploy`.
```

Comment:
Note: In order to deploy to the same stack in production, we would need to have the same samconfig.toml?

Reply:
Can add that, I suppose should also include like "production samconfig.toml is stored in 1Password (or somewhere)"? Or how to verify ppl have the same samconfig.toml?

Reply:
1Password would be good approach. I created 1Password - Aiven DevPortal and included the env.production.json

Reply:
Added

Reply:
I think having the same setup is right - but nobody should connect their dev platform to the live storage, so we can maybe put the toml in the repo with placehodlers, and (as we had before) add instructions for creating the data structures etc. They're just Aiven services, we can easily spin up new ones to use in development.

Reply:
This is for deploying the function to AWS Lambda for production. Not needed for local development.

Reply:
Technically in local if one wants to upload a function to a personal account (or use Aiven development account) then it also uses the same `samconfig.toml` file. Perhaps there's some way to create a `samconfig.prod.toml` and `samconfig.dev.toml` and use accordingly.